crates/driver/src/domain/quote.rs

```diff
@@ -87,7 +87,12 @@ impl Order {
         };
 
         let solutions = solver
-            .solve(&self.fake_auction(eth, tokens).await?, &liquidity)
+            .solve(
+                &self
+                    .fake_auction(eth, tokens, solver.quote_using_limit_orders())
```

Comment:
This call is getting a bit weird. Maybe the code would be slightly nicer if `fake_auction` would be a function on the solver instead that takes the order? 🤔

Reply:
I expect `quote_using_limit_orders` to be removed once all solvers are able to quote using limit orders, which would mean we would be left without the only reason to put the `fake_auction` on `solver` if we do what you suggest.

Reply:
Yeah, fair point. Overall I find how the quoting code is currently structured a bit weird regardless but this could then be a refactoring outside of this PR.